### PR TITLE
[Design System] Iterate Button styles

### PR DESCRIPTION
**Pull request by MildTomato**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

# waiting for project approval

In draft, waiting for [Project approval](https://www.notion.so/supabase/Design-system-Button-styling-iteration-a47e3e6c91a64b8e9b43a2945fc6a54b?pvs=4)

YES/NO

- icons now colored by default
- primary button a11y contrast increased
- docs updated

### CODE REVIEW

**Comment by MildTomato** [comment]
In draft, waiting for Project approval
https://www.notion.so/supabase/Design-system-Button-styling-iteration-a47e3e6c91a64b8e9b43a2945fc6a54b?pvs=4

**Review by alaister — APPROVED** [review]

**Pull request merged**
